# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/base/nsCaret.h
# Commit: a7a2744670a7
# Full Hash: a7a2744670a7ce15ccafd8c59b909d90b8e2837a
# Author: Sean Feng <sefeng@mozilla.com>
# Date: 2024-01-11 09:52:49
# Regressor Bug: 1860328
# File Overlap Count: 2
# Description:
#   Bug 1860328 - Fix a bug where caret misses when dragging and dropping r=emilio
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D192377
# ==============================================================================

diff -r 62230fd018f6 -r a7a2744670a7 layout/base/nsCaret.h
--- a/layout/base/nsCaret.h	Wed Jan 10 16:50:49 2024 +0000
+++ b/layout/base/nsCaret.h	Wed Jan 10 17:03:38 2024 +0000
@@ -12,6 +12,7 @@
 #include "mozilla/MemoryReporting.h"
 #include "mozilla/dom/Selection.h"
 #include "nsCoord.h"
+#include "nsIFrame.h"
 #include "nsISelectionListener.h"
 #include "nsIWeakReferenceUtils.h"
 #include "nsPoint.h"
@@ -196,6 +197,10 @@
 
   size_t SizeOfIncludingThis(mozilla::MallocSizeOf aMallocSizeOf) const;
 
+  void SetLastCaretFrame(nsIFrame* aLastCaretFrame) {
+    mLastCaretFrame = aLastCaretFrame;
+  }
+
  protected:
   static void CaretBlinkCallback(nsITimer* aTimer, void* aClosure);
 
@@ -279,6 +284,11 @@
    * it's in non-user-modifiable content.
    */
   bool mIgnoreUserModify;
+
+  /**
+   * mLastCaretFrame is the last caret frame that was scheduled to paint.
+   */
+  WeakFrame mLastCaretFrame;
 };
 
 #endif  // nsCaret_h__